src/core/components/Search/SearchResults.js

```diff
@@ -52,23 +71,30 @@ class SearchResults extends React.Component {
           ))}
         </ul>
       );
-    } else if (query && loading) {
+    } else if (searchParamIsPresent && loading) {
       messageText = i18n.gettext('Searching...');
-    } else if (query && results.length === 0) {
+    } else if (!loading && query && count === 0) {
       messageText = i18n.sprintf(
         i18n.gettext('No results were found for "%(query)s".'), { query });
-    } else if (query !== null) {
-      messageText = i18n.gettext('Please supply a valid search');
+    } else if (!loading && searchParamIsPresent && count === 0) {
+      // TODO: Add the extension type, if available, so it says "no extensions"
+      // found that match your search or something.
+      messageText = i18n.gettext('No results were found.');
+    } else if (!searchParamIsPresent && !loading) {
+      messageText = i18n.gettext(
+        "Please enter search terms to search all of Mozilla's Add-ons.");
     }
 
-    const message = messageText ?
```

Comment:
`messageText` is always set, so right now it is never `null` the code coverage not minding the line being removed proves this.

I could add it back in but there's no code path that hits it now.